includes/frontend/class-wcmp-checkout.php

```diff
@@ -415,20 +415,20 @@ public static function save_delivery_options($order_id)
      */
     private function getShippingMethodsAllowingDeliveryOptions(): array
     {
-        $allowedMethods = [];
-        $displayFor     = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_DELIVERY_OPTIONS_DISPLAY);
+        $allowedMethods               = [];
+        $displayFor                   = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_DELIVERY_OPTIONS_DISPLAY);
+        $shippingMethodsByPackageType = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_SHIPPING_METHODS_PACKAGE_TYPES);
 
-        if (WCMP_Settings_Data::DISPLAY_FOR_ALL_METHODS === $displayFor) {
+        if (WCMP_Settings_Data::DISPLAY_FOR_ALL_METHODS === $displayFor || ! $shippingMethodsByPackageType) {
             return $allowedMethods;
         }
 
-        $shippingMethodsByPackageType = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_SHIPPING_METHODS_PACKAGE_TYPES);
-        $shippingMethodsForPackage    = $shippingMethodsByPackageType[AbstractConsignment::PACKAGE_TYPE_PACKAGE_NAME];
+        $shippingMethodsForPackage = $shippingMethodsByPackageType[AbstractConsignment::PACKAGE_TYPE_PACKAGE_NAME];
```

Comment:
Ik had nog even verder gekeken, maar deze kan toch nog steeds NULL worden in sommige gevallen (als de AbstractConsignment::PACKAGE_TYPE_PACKAGE_NAME niet in de array zit om wat voor reden dan ook) en tot dezelfde fout leiden bij de foreach eronder? Moet die dan ook niet naar voren getrokken worden en meegenomen worden in de premature return $allowedMethods? Of weten we 100% zeker dat die key altijd in de array bestaat, dat blijkt niet echt uit de code, volgens mij.

Reply:
De als de`$shippingMethodsByPackageType` gevuld is, kan je er altijd vanuit gaan dat `package` erin zit. Dit komt omdat wij kijken naar de setting `shipping_methods_package_types` waarin het volgende zal zitten:

![image](https://user-images.githubusercontent.com/11389731/114880031-ffe0a380-9e01-11eb-8dee-6c1c7c778be6.png)

Reply:
En omdat we de delivery options alleen bij `package` willen laten zien, benoem je die specifiek key.

Reply:
my bad, duidelijk ja

Reply:
Geen probleem, hierdoor blijf ik ook scherp 😄